docs/src/literate-tutorials/linear_elasticity.jl

```diff
@@ -1,3 +1,354 @@
+# \end{align*}
+# ```
+#
+# Starting from Young's modulus, $E$, and Poisson's ratio, $\nu$, the shear and bulk modulus are
+# ```math
+# G = \frac{E}{2(1 + \nu)}, \quad K = \frac{E}{3(1 - 2\nu)}
+# ```
+# Finally, the stiffness tensor can be calculated as the derivative of stress wrt. strain.
+# Due to the linearity, we can calculate this at any point, and choose zero strain.
+Emod = 200e3 # Young's modulus [MPa]
+ν = 0.3      # Poisson's ratio [-]
+
+Gmod = Emod / (2(1 + ν))  # Shear modulus
+Kmod = Emod / (3(1 - 2ν)) # Bulk modulus
+E4 = gradient(ϵ -> 2 * Gmod * dev(ϵ) + 3 * Kmod * vol(ϵ), zero(SymmetricTensor{2,2}));
+
+#md # !!! details "Plane stress instead of plane strain?"
+#md #     In order to change this tutorial to consider plane stress instead of plane strain,
+#md #     the stiffness tensor should be changed to reflect this. The plane stress elasticity
+#md #     stiffness matrix in Voigt notation for engineering shear strains, is given as
+#md #     ```math
+#md #     \underline{\underline{\boldsymbol{E}}} = \frac{E}{1 - \nu^2}\begin{bmatrix}
+#md #     1 & \nu & 0 \\
+#md #     \nu & 1 & 0 \\
+#md #     0 & 0 & (1 - \nu)/2
+#md #     \end{bmatrix}
+#md #     ```
+#md #     This matrix can be converted into the 4th order stiffness tensor as
+#md #     ```julia
+#md #     E_voigt = Emod * [1.0 ν 0.0; ν 1.0 0.0; 0.0 0.0 (1-ν)/2] / (1 - ν^2)
+#md #     E4 = fromvoigt(SymmetricTensor{4,2}, E_voigt)
+#md #     ```
+#md #
+# ### Element routine
+# To calculate the global stiffness matrix, $K_{ij}$, the element routine computes the
+# local stiffness matrix `ke` for a single element and assembles it into the global matrix.
+# `ke` is pre-allocated and reused for all elements.
+#
+# Note that the elastic stiffness tensor $\boldsymbol{\mathsf{E}}$ is constant.
+# Thus is needs to be computed and once and can then be used for all integration points.
+function assemble_cell!(ke, cellvalues, ∂σ∂ε)
+    for q_point in 1:getnquadpoints(cellvalues)
+        ## Get the integration weight for the quadrature point
+        dΩ = getdetJdV(cellvalues, q_point)
+        for i in 1:getnbasefunctions(cellvalues)
+            ## Gradient of the test function
+            ∇Nᵢ = shape_gradient(cellvalues, q_point, i)
+            for j in 1:getnbasefunctions(cellvalues)
+                ## Symmetric gradient of the trial function
+                ∇ˢʸᵐNⱼ = shape_symmetric_gradient(cellvalues, q_point, j)
+                ke[i, j] += (∇Nᵢ ⊡ ∂σ∂ε ⊡ ∇ˢʸᵐNⱼ) * dΩ
+            end
+        end
+    end
+    return ke
+end
+#md nothing #hide
+#
+#md # !!! details "Performance tips"
+#md #     1. In this tutorial, `∂σ∂ε`, is minor symmetric, and we could have used
+#md #        the symmetric part of test function gradient above.
+#md #     2. The product `∇Nᵢ ⊡ ∂σ∂ε` can be done outside the "j"-loop, but is kept inside here
+#md #        to highlight the similarity to the finite element form.
+#md #
```

Comment:
3. We can eliminate the fourth order tensor fully here if necessary (in the linear elastic model).

Reply:
Do you mean by calculating the volumetric and deviatoric terms separately? At least when I tried once, that was slower IIRC.

Reply:
I would argue the performance tips should all have been tested to see that they actually make a difference. Otherwise they are just noisy guesses.

Reply:
Can we add the benchmarks then? And also the hint that users shall not compute them separately (as this is also counterintuitive).

Reply:
I've removed performance tips completely now. I don't think this is required for the introductory tutorial. Originally, I added the one about symmetric part in response to https://github.com/Ferrite-FEM/Ferrite.jl/pull/799#discussion_r1378740246, and then the hoisting of computation felt that it should also be mentioned.

Thinking more about it, this would rather fit in a separate performance tip how-to / advice section in the docs.

Reply:
xref https://github.com/Ferrite-FEM/Ferrite.jl/issues/1023 for later